### PR TITLE
Add images for Java 21

### PR DESCRIPTION
**Pull request by kamperiadis**
- Add v4 images for Java 21
- Update files to include Java 21:
  - core-tools-publish.yml
  - java-build.yml
  - publish-appservice-stage-sovereign-clouds.yml
  - publish-appservice-stage.yml
  - publish.yml
  - republish.xml

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [] If applicable, the PR references the bug/issue that it fixes in the description.
- [] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

**Review by kaibocai — DISMISSED** [review]
LGTM thank you!

**Review by CooperLink — CHANGES_REQUESTED** [review]
Please add the new image to our republish pipeline : https://github.com/Azure/azure-functions-docker/blob/dev/host/4/republish.yml

**Review by CooperLink — DISMISSED** [review]
LGTM

File: host/4/bullseye/amd64/java/java21/java21-build.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+FROM mcr.microsoft.com/dotnet/sdk:6.0
```
Comment (pragnagopa) [inline host/4/bullseye/amd64/java/java21/java21-build.Dockerfile]: as bulleseye will be retiring soon, new images need to be on book work.

@CooperLink - is that possible?

**Review by pragnagopa — CHANGES_REQUESTED** [review]
Need to discuss how to move to bookwork

File: host/4/bullseye/amd64/java/java21/java21-composite.template [hunk 2]
```diff
@@ -0,0 +1,38 @@
+ARG JAVA_VERSION=21.0.0
+ARG JAVA_HOME=/usr/lib/jvm/msft-21-x64
+FROM mcr.microsoft.com/dotnet/runtime-deps:6.0
+ARG HOST_VERSION
+ARG JAVA_VERSION
```
Comment (CooperLink) [inline host/4/bullseye/amd64/java/java21/java21-composite.template]: Currently the host composite image is based on Bullseye so for now it likely makes more sense to replace this with an -appservice image. I can show Kirstyn how to do this. 
This is necessary to get Host on bookworm
@pragnagopa

**Review by pragnagopa — DISMISSED** [review]

**Review by CooperLink — APPROVED** [review]

**Pull request merged**